EasyPost/PaymentMethod.cs

```diff
@@ -1,38 +1,100 @@
+        /// </summary>
+        public enum Priority
+        {
+            Primary,
+            Secondary
+        }
+
         [JsonProperty("id")]
         public string id { get; set; }
         [JsonProperty("object")]
         public string Object { get; set; }
         [JsonProperty("primary_payment_method")]
-        public CreditCard primary_payment_method { get; set; }
+        public PaymentMethodObject primary_payment_method { get; set; }
         [JsonProperty("secondary_payment_method")]
-        public CreditCard secondary_payment_method { get; set; }
+        public PaymentMethodObject secondary_payment_method { get; set; }
+    }
+
+    /// <summary>
+    ///     Represents a credit card or a bank account.
+    /// </summary>
+    public class PaymentMethodObject : Resource
+    {
+        public enum PaymentMethodType
+        {
+            CreditCard,
+            BankAccount
+        }
+
+        [JsonProperty("bank_name")]
+        public string bank_name { get; set; }
+        [JsonProperty("brand")]
+        public string brand { get; set; }
+        [JsonProperty("country")]
+        public string country { get; set; }
+        [JsonProperty("disabled_at")]
+        public string disabled_at { get; set; }
+        [JsonProperty("exp_month")]
+        public string exp_month { get; set; }
+        [JsonProperty("exp_year")]
+        public string exp_year { get; set; }
+        [JsonProperty("id")]
+        public string id { get; set; }
+        [JsonProperty("last4")]
+        public string last4 { get; set; }
+        [JsonProperty("name")]
+        public string name { get; set; }
+        [JsonProperty("object")]
+        public string Object { get; set; }
 
         /// <summary>
-        ///     List all payment methods for this account.
+        ///     Get what type of payment method this is (credit card, bank account, etc.)
         /// </summary>
-        /// <returns>An EasyPost.PaymentMethod summary object.</returns>
-        /// <exception cref="Exception"></exception>
-        public static async Task<PaymentMethod> All()
+        public PaymentMethodType? Type
         {
-            Request request = new Request("payment_methods", Method.Get);
+            get
+            {
+                if (id == null)
+                {
+                    return null;
+                }
 
-            PaymentMethod paymentMethod = await request.Execute<PaymentMethod>();
+                if (id.StartsWith("card_"))
+                {
+                    return PaymentMethodType.CreditCard;
+                }
 
-            if (paymentMethod.id == null)
-            {
-                throw new Exception("Billing has not been setup for this user. Please add a payment method.");
+                if (id.StartsWith("bank_"))
+                {
+                    return PaymentMethodType.BankAccount;
+                }
+
+                return null;
             }
+        }
+
+        [JsonProperty("verified")]
```

Comment:
Why is this JsonProperty isolated from above?

Reply:
Should we move this to the top?

Reply:
The getter/setter properties are sorted alphabetically, so the `Type` property is being organized above this, unfortunately. I believe our linter will complain if I were to force it to the top.

Reply:
Oh my... please look into fixing this. I don't want to propagate bad patterns like this.

Reply:
There's nothing we can do, since it's all getter/setters

Reply:
Resolved using [region markers ](https://docs.microsoft.com/en-us/dotnet/visual-basic/language-reference/directives/region-directive) to keep elements grouped together, even during linting

Reply:
Looove this! Thanks for going the extra mile to correct this.